test/tools/d_do_test.d

```diff
@@ -835,6 +840,17 @@ int tryMain(string[] args)
             writefln("Test %s/%s.%s failed.  The logged output:", input_dir, test_name, test_extension);
             writeln(output_file.readText);
             output_file.remove();
+
+            // automatically rerun a segfaulting test and print its stack trace
+            version(linux)
```

Comment:
Why only Linux?

Reply:
Because I didn't know whether it would work on other platforms and I thought it's good to start with something that I can test. Also most of our CIs are Linux ones, so it would already have a measurable impact.
IIRC OSX uses `lldb` and Windows has who knows what debugger available?

(+I didn't feel like fighting the auto-tester for something that hasn't received feedback yet.)